pkg/iris/mapping.go

```diff
@@ -34,45 +41,102 @@ func NewMapper(
-		deployments, err := m.KubernetesApi.Deployments(namespace.Name)
-		if err != nil {
-			return nil, err
-		}
-		log.Infof("mapping deployments for namespace [%s]", namespace.Name)
-		mappedDeployments := m.MapDeployments(deployments)
-		allDiscoveryItems = append(allDiscoveryItems, mappedDeployments...)
+	k8sVersion := set.NewStringSet()
+	osImage := set.NewStringSet()
+
+	for _, n := range items {
+		k8sVersion.Add(n.Status.NodeInfo.KubeletVersion)
+		osImage.Add(n.Status.NodeInfo.OSImage)
+
 	}
-	log.Info("Fetching deployments for namespaces completed")
-	return allDiscoveryItems, nil
+	return ClusterDTO{
+		name:       clusterName,
+		k8sVersion: strings.Join(k8sVersion.Items(), ", "),
+		nodesCount: len(items),
+		osImage:    strings.Join(osImage.Items(), ", "),
+	}, nil
 }
 
-func (m *mapper) MapDeployments(deployments *appsv1.DeploymentList) []DiscoveryItem {
-	var deploymentDiscoveryItems []DiscoveryItem
+func (m *mapper) GetDeployments(deployments *appsv1.DeploymentList, services *v1.ServiceList) ([]models.Deployment, error) {
+	var allDeployments []models.Deployment
+
 	for _, deployment := range deployments.Items {
-		deploymentItem := NewDeploymentEvent(*m, deployment)
-		softwareArtifactItem := NewSoftwareArtifactEvent(*m, deployment)
-		deploymentDiscoveryItems = append(deploymentDiscoveryItems, *deploymentItem, *softwareArtifactItem)
+		deploymentService := ""
+		// Check if any service has the exact same selector labels and use this as the service related to the deployment
+		deploymentService = ResolveServiceForDeployment(services, deployment)
+
+		mappedDeployment := models.Deployment{
+			Service: models.Service{
+				Name: deploymentService,
+			},
+			Name:      deployment.Name,
+			Labels:    deployment.ObjectMeta.Labels,
+			Timestamp: deployment.CreationTimestamp.UTC().Format(time.RFC3339),
+			Properties: models.Properties{
+				UpdateStrategy: string(deployment.Spec.Strategy.Type),
+				Replicas:       strconv.FormatInt(int64(deployment.Status.Replicas), 10),
+				K8sLimits:      CreateK8sResources(deployment.Spec.Template.Spec.Containers[0].Resources.Limits),
+				K8sRequests:    CreateK8sResources(deployment.Spec.Template.Spec.Containers[0].Resources.Requests),
+			},
+		}
+		allDeployments = append(allDeployments, mappedDeployment)
+	}
+
+	return allDeployments, nil
+}
+
+func CreateK8sResources(resourceList v1.ResourceList) models.K8sResources {
+	cpu := resourceList[v1.ResourceCPU]
+	cpuString := ""
+	if !cpu.IsZero() {
+		cpuString = cpu.String()
+	}
+
+	memory := resourceList[v1.ResourceMemory]
+	memoryString := ""
+	if !memory.IsZero() {
+		memoryString = memory.String()
+	}
+
+	return models.K8sResources{
+		Cpu:    cpuString,
+		Memory: memoryString,
 	}
-	return deploymentDiscoveryItems
 }
 
-func blacklistedNS(a string, list []string) bool {
-	for _, b := range list {
-		if b == a {
-			return true
+func ResolveServiceForDeployment(services *v1.ServiceList, deployment appsv1.Deployment) string {
```

Comment:
Do I undestand this correctly that using the old k8s connector terminology we are discoverying software artifacts based on labels instead of namespace? Or is this something different entirely?